posix/ioctl.c

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/stdarg.h>
+#include <openenclave/internal/posix/device.h>
+#include <openenclave/internal/posix/fdtable.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/trace.h>
+
+int __oe_ioctl(int fd, unsigned long request, uint64_t arg)
+{
+    int ret = -1;
+    static const unsigned long _TIOCGWINSZ = 0x5413;
+
+    if (request == _TIOCGWINSZ)
```

Comment:
This looks like a placeholder, can we at least add an issue to track if this is going to be fully implemented?

- This handles TIOCGWINSZ but not TIOCSWINSZ (set)
- This implementation just assumes a fixed default terminal size.

It seems like this should just marshal through to host via the same ioctl call and let a windows specific impl handle it on host side instead of short-circuiting it here with this stub impl.

Reply:
Thanks. Created issue #1808 to track. I also moved this block to consolefs.c.